### ISSUE TITLE
question

### ISSUE THREAD

**Issue by lainanhui** [bdi-lab/InGram#11]
Hello, thank you for writing this paper, but I don't particularly understand inductive knowledge graph embedding, I would like to ask that every Epoch performs the initialization of entities and relations, my understanding is that in order to make every entity and relation to learn as if it is newly emerged, so every Epoch initializes the entities and relations, I would like to ask if I have understood it correctly. Another question is, why do we need to divide the training set?
Thank you author, I hope everything goes well and I look forward to your reply!

**Comment by jaejunlee714** [comment]
Hi, thanks for your interest in our work. As you understood, the initialization is performed so that the entities and relations are treated as newly emerged ones.
Since InGram learns to generate embeddings from a given knowledge graph, and then predict the missing links in that graph, we need a knowledge graph to generate embeddings (F_tr), and some triplets(T_tr) to calculate the loss. Therefore, we divide the training set. Since using the same F_tr and T_tr every epoch might lead to overfitting, we used the dynamic split strategy that re-splits E_tr to F_tr and T_tr every epoch.

**Comment by lainanhui** [comment]
Thanks for your reply! Sorry, my previous understanding is that traditional knowledge graphs do an optimization directly in the information graph after obtaining the embedding, I would like to ask why inductive knowledge graphs need another graph for optimization?
Thanks, looking forward to your reply!

**Comment by jaejunlee714** [comment]
Since the triplets in the F_tr (graph for generating embeddings) are already known to that embedding, using the triplets in F_tr for optimization is too easy for the model - the model already knows the answer. Some models like RED-GNN, NeuralLP, or DRUM also splits the train set and the fact set. If we use the same triplets for embedding generation and prediction, the model might only learn to retrieve.
Note that this does not hold for knowledge graph embedding methods like RotatE, TransE, ANALOGY, etc, as what they only learn is the embeddings directly, and do not use some portion of the training set to generate the embeddings.

**Comment by lainanhui** [comment]
Thanks for your reply, now I understand the inductive knowledge graph embedding is to generate the target embedding by the known embedding, but the known embedding should be obtained after training, if so, how to get this known embedding?
Sorry for my questions, looking forward to your reply!

**Comment by lainanhui** [comment]
Thank you for your reply, I would like to ask is it right to generate the embedding of the target entity from the known embedding and the relationship graph, then this known embedding is obtained by training? Because I see that the code initializes all embedding, so I don't know if this known embedding is trained or also randomly initialized?
Sorry I'm a bit full of questions, looking forward to your reply!

**Comment by jaejunlee714** [comment]
InGram assumes that all entities and relations are unknown. So, even if the entities and relations are already seen in the training set, they are randomly initialized.

**Comment by lainanhui** [comment]
Thank you for your reply! I wish you all the best!
